airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/java/io/airbyte/integrations/destination/bigquery/formatter/arrayformater/ModernArrayFormatter.java

```diff
@@ -0,0 +1,80 @@
+package io.airbyte.integrations.destination.bigquery.formatter.arrayformater;
+
+import static io.airbyte.integrations.destination.bigquery.formatter.DefaultBigQueryDenormalizedRecordFormatter.PROPERTIES_FIELD;
+import static io.airbyte.integrations.destination.bigquery.formatter.DefaultBigQueryDenormalizedRecordFormatter.TYPE_FIELD;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldList;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.util.MoreIterators;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ModernArrayFormatter implements ArrayFormatter {
```

Comment:
nit: I don't think `Modern` adds value as a description here. Please rename to `DefaultArrayFormatter`